powdr/src/lib.rs

```diff
@@ -125,7 +125,12 @@ impl Session {
         }
     }
 
-    pub fn write<S: serde::Serialize>(self, channel: u32, data: &S) -> Self {
+    pub fn write<S: serde::Serialize>(&mut self, data: &S) {
```

Comment:
a bit annoying that this function doesn't take `self` and return `Self`, before this was nicer cus you could do
```rust
let session = Session::builder()
    ...
    .build()
    .write(...)
    .write(...);
```

With this we need to do
```rust
let mut session = Session::builder()
    ...
    .build();
session.write(...)
session.write(...);
```
We could keep the same interface, but then we'll need to `mem::take` `self.pipeline` here, change it, and make a new Session out of that. Kinda ugly too, we need to decide what's uglier.

Reviewer, lmk your opinion.

Reply:
why the need to mem::take the pipeline? should work fine if you take `mut self` no?

Reply:
done